test/unit/calculators/checks_benefits_support_calculator_test.rb

```diff
@@ -221,473 +335,709 @@ class CheckBenefitsSupportCalculatorTest < ActiveSupport::TestCase
+          should "be false is country is Wales working over 16 hours withour child aged 3 to 4" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.where_do_you_live = "wales"
+            calculator.are_you_working = "yes_over_16_hours_per_week"
+            calculator.children_living_with_you = "yes"
+            calculator.age_of_children = "1,5_to_11"
+            assert_not calculator.eligible_for_childcare_3_4yr_olds_wales?
+          end
         end
       end
 
-      context "#eligible_for_30hrs_free_childcare_3_4yrs?" do
-        should "return true if eligible for 30 Hours Free Childcare for 3 and 4 Year Olds" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.where_do_you_live = "england"
-          %w[yes_over_16_hours_per_week yes_under_16_hours_per_week].each do |working_hours|
-            calculator.are_you_working = working_hours
+      context "#eligible_for_15hrs_free_childcare_3_4yr_olds?" do
+        context "when eligible" do
+          should "be true if country is England, with child aged 3 to 4" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.where_do_you_live = "england"
             calculator.children_living_with_you = "yes"
-            calculator.age_of_children = "3_to_4"
-            assert calculator.eligible_for_30hrs_free_childcare_3_4yrs?
+            calculator.age_of_children = "1,3_to_4"
+            assert calculator.eligible_for_15hrs_free_childcare_3_4yr_olds?
           end
         end
 
-        should "return false if not eligible for 30 Hours Free Childcare for 3 and 4 Year Olds - not working" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.where_do_you_live = "england"
-          calculator.are_you_working = "no"
-          calculator.children_living_with_you = "yes"
-          calculator.age_of_children = "3_to_4"
-          assert_not calculator.eligible_for_30hrs_free_childcare_3_4yrs?
-        end
+        context "when ineligible" do
+          should "be false if country is not England with child aged 3 to 4" do
+            %w[scotland wales northern-ireland].each do |country|
+              calculator = CheckBenefitsSupportCalculator.new
+              calculator.where_do_you_live = country
+              calculator.children_living_with_you = "yes"
+              calculator.age_of_children = "1,3_to_4"
+              assert_not calculator.eligible_for_15hrs_free_childcare_3_4yr_olds?
+            end
+          end
 
-        should "return false if not eligible for 30 Hours Free Childcare for 3 and 4 Year Olds - no children" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.where_do_you_live = "england"
-          %w[yes_over_16_hours_per_week yes_under_16_hours_per_week].each do |working_hours|
-            calculator.are_you_working = working_hours
+          should "be false if country is England without" do
```

Comment:
"be false if country is England without" -> "be false if country is England without children"